WalletWasabi/WabiSabi/Backend/Rounds/CoinJoinStorage/ICoinJoinIdStore.cs

```diff
@@ -7,4 +8,6 @@ public interface ICoinJoinIdStore
 	bool TryAdd(uint256 id);
 
 	bool Contains(uint256 id);
+
+	ReadOnlyDictionary<uint256, uint> GetCoinJoinsIdsWithOrder();
```

Comment:
I would return an array instead of a dictionary with order.

```suggestion
	ImmutableArray<uint256> GetCoinJoinsIds();
```

Reply:
I'm doing that to preserve the performances of a `Dictionary.ContainsKey` over `Array.Contains`, as I'm worried about the complexity of the `CoinJoinMempoolManager.Mempool_Tick` method. With this extra context, do you still think that it's better to use an `ImmutableArray`?